GVFS/GVFS/CommandLine/ConfigVerb.cs

```diff
@@ -10,18 +11,34 @@ public class ConfigVerb : GVFSVerb.ForNoEnlistment
         private const string ConfigVerbName = "config";
         private LocalGVFSConfig localConfig;
 
+        [Option(
+            "list",
+            SetName = "needNoKeys",

```

Comment:
SetName help us specify mutually exclusive options. 

Example: if the user tries ```gvfs config --list --delete <key>``` parser displays the following error. 
<img width="756" alt="screen shot 2018-10-15 at 12 20 25 pm" src="https://user-images.githubusercontent.com/378580/46979425-97cd6d00-d09f-11e8-99df-ca1ece5bada8.png">

In this case it helps avoid chaining multiple Options in a single command. But this is more useful when there are competing Options for a given command line Value. For example --delete Option needs a key Value. Suppose there was an Option --set which also needs a key Value. And user types ```gvfs config --delete --set foo bar```. SetName helps avoid situations like this.

Reply:
Yep I get the value of having parameter sets. I was more asking about this particular API. E.g. How is the name used? How does it handle overlapping parameter sets? How cleanly would it apply to the PrefetchVerb?

Parts of this API are not fully fleshed out and are still buggy, so I wanted to better understand how complete of an implementation this is before we decide to use it. I think it would be instructive to go ahead and update PrefetchVerb with this pattern, because that verb has more complex requirements and will help us flush out any issues.